base/db/R/dbfiles.R

```diff
@@ -109,7 +111,13 @@ dbfile.input.insert <- function(in.path, in.prefix, siteid, startdate, enddate,
   if (is.null(inputid)) {
     # Either there was no existing input, or there was but the dates don't match and
     # allow.conflicting.dates==TRUE. So, insert new input record.
-    if (parent == "") {
+    # adding is.null(startdate) to add inputs like soil that don't have dates
+    if(parent == "" && is.null(startdate)) { 
```

Comment:
This logic looks like it would miss the case where startdate is null but parent isn't blank